packages/core/strapi/lib/services/entity-validator/validators.js

```diff
@@ -20,69 +26,107 @@ const composeValidators = (...fns) => (attr, { isDraft }) => {
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinLengthValidator = ({ minLength }, validator, { isDraft }) =>
+const addMinLengthValidator = (validator, { minLength }, { isDraft }) =>
   _.isInteger(minLength) && !isDraft ? validator.min(minLength) : validator;
 
 /**
  * Adds maxLength validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxLengthValidator = ({ maxLength }, validator) =>
+const addMaxLengthValidator = (validator, { maxLength }) =>
   _.isInteger(maxLength) ? validator.max(maxLength) : validator;
 
 /**
  * Adds min integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinIntegerValidator = ({ min }, validator) =>
+const addMinIntegerValidator = (validator, { min }) =>
   _.isNumber(min) ? validator.min(_.toInteger(min)) : validator;
 
 /**
  * Adds max integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxIntegerValidator = ({ max }, validator) =>
+const addMaxIntegerValidator = (validator, { max }) =>
   _.isNumber(max) ? validator.max(_.toInteger(max)) : validator;
 
 /**
  * Adds min float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinFloatValidator = ({ min }, validator) =>
+const addMinFloatValidator = (validator, { min }) =>
   _.isNumber(min) ? validator.min(min) : validator;
 
 /**
  * Adds max float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxFloatValidator = ({ max }, validator) =>
+const addMaxFloatValidator = (validator, { max }) =>
   _.isNumber(max) ? validator.max(max) : validator;
 
 /**
  * Adds regex validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addStringRegexValidator = ({ regex }, validator) =>
+const addStringRegexValidator = (validator, { regex }) =>
   _.isUndefined(regex) ? validator : validator.matches(new RegExp(regex));
 
+const addUniqueValidator = (validator, attr, __, model, updatedAttribute, entity) => {
+  if (!attr.unique && attr.type !== 'uid') {
+    return validator;
+  }
+
+  /**
+   * If the attribute value is `null` we want to skip the unique validation.
+   * Otherwise it'll only accept a single `null` entry in the database.
+   */
+  if (updatedAttribute.value === null) {
+    return validator;
+  }
+
+  /**
+   * If the attribute is unchanged we skip the unique verification. This will
+   * prevent the validator to be triggered in case the user activated the
+   * unique constraint after already creating multiple entries with
+   * the same attribute value for that field.
+   */
+  if (entity && updatedAttribute.value === entity[updatedAttribute.name]) {
+    return validator;
+  }
```

Comment:
These conditions could still be tested in the test('unique' .... as they are part of the test in itself I think